.eslintignore

```diff
@@ -1,2 +1,4 @@
 dist/**
-.github/**
+yarn.lock
+demos/**
+**/*.json
```

Comment:
Why are you ignoring json files ?
```suggestion
```

Reply:
eslint+lint-staged think package.json is invalid and refuse commit.
<img width="419" alt="Screen Shot 2021-12-10 at 8 07 36 PM" src="https://user-images.githubusercontent.com/76864176/145571883-e14fed6e-aa05-40a1-92c8-bcb01f517974.png">

Reply:
Then let's just ignore package.json and not all the json files

Reply:
In fact, this happens on all json files more than just package.json. For example, I changed cypress.json and add it to stage and exec `yarn lint-staged`
<img width="420" alt="Screen Shot 2021-12-10 at 8 17 14 PM" src="https://user-images.githubusercontent.com/76864176/145573142-897fac71-c837-4af1-928a-57ace1321ca8.png">

But it doesn't throw error when just exec `yarn lint`

Reply:
What is causing this the lint-staged configuration or the eslint configuration?

Reply:
In my understanding, ESLint is a javascript linter which is used to lint js/ts/jsx. When exec `yarn eslint .`, which means lint all js files in project. But when ESLint is used with lint-staged, lint-staged will provides it with other types of files, the parser '@babel/eslint-parser' can not work with json.

And I noticed that jsx was enabled in parserOptions, although I didn't find anywhere to use it. Maybe set .lintstagedrc.json with

``` json
{
  "*.{js,jsx}": ["yarn lint:fix", "git add"],
  "*.{json,md,css,scss,html,yml}": ["prettier --write", "git add"]
}
```

Reply:
lint staged files:
- js and jsx: lint by ESLint
- json,md,css,scss,html,yml: format by prettier